drivers/pwm/pwm_stm32.c

```diff
@@ -270,9 +278,17 @@ static int pwm_stm32_pin_set(const struct device *dev, uint32_t pwm,
 		LL_TIM_OC_StructInit(&oc_init);
 
 		oc_init.OCMode = LL_TIM_OCMODE_PWM1;
-		oc_init.OCState = LL_TIM_OCSTATE_ENABLE;
+
+		/* the flags holds the PWM_COMPLEMENTARY information */
+		if ((flags & PWM_COMPLEMENTARY_MASK) == PWM_COMPLEMENTARY) {
```

Comment:
An error could be raised in case channel doesn't have complementary capability, but it is set in DTS.

Reply:
@ABOSTM This is not possible as is today and proposed pinctrl bondings evolutions to enable it were rejected by mainatainer. This is specified as a "user's responsibility" to provide correct pinctrl configuration.

Reply:
I guess capability is known, however, whether pin is correctly configured is not.

Reply:
I mean runtime error thanks to macro IS_TIM_COMPLEMENTARY_CHANNELS()